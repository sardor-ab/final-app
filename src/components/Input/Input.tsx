import { PureComponent } from "react";

export class Input extends PureComponent {
  render() {
    console.log("first");
    return <input type="text" />;
  }
}
