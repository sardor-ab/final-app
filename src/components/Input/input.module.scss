div.select_input {
  padding: 15px 0;
  position: relative;

  label {
    position: absolute;
    top: 0px;
    left: 12px;
    font-size: 12px;
    user-select: none;
    pointer-events: none;
    background-color: #ffffff;
    -webkit-transform: translate(0px, 10px) scale(1);
    -moz-transform: translate(0px, 10px) scale(1);
    -ms-transform: translate(0px, 10px) scale(1);
    transform: translate(0px, 10px) scale(1);
  }

  input {
    width: calc(100% - 30px);
    border: 1px solid #333333;
    border-radius: 4px;
    background-color: #ffffff;
    font-weight: 400;
    font-size: 14px;
    line-height: 19px;
    padding: 12px 0 12px 12px;
    outline: none;
  }

  span {
    font-weight: 400;
    font-size: 0.75rem;
    line-height: 1.66;
    letter-spacing: 0.03333em;
    text-align: left;
    margin: 3px 10px 0px;
  }
}
