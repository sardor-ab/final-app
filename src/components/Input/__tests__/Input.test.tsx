import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Input } from '../Input';

describe('Input', () => {
    test('Снимок input-a', () => {
        const { getByTestId } = render(<Input onChange={console.log} value="value" />);

        expect(getByTestId('input')).toMatchSnapshot();
    });

    test('Если есть helperText, то строка должна отобразиться в DOM', () => {
        const { getByTestId } = render(
            <Input helperText="helperText" onChange={console.log} value="value" />
        );

        expect(getByTestId(/helperText/)).toMatchSnapshot();
    });

    test('Если есть label, то строка должна отобразиться в DOM', () => {
        const { getByTestId } = render(
            <Input onChange={console.log} label="label" value="value" />
        );

        expect(getByTestId('input')).toMatchSnapshot();
    });

    test('Если есть onClear, то крестик должен отобразиться в DOM', () => {
        const { getByTestId } = render(
            <Input onChange={console.log} onClear={jest.fn()} value="value" />
        );

        expect(getByTestId('onClearButton')).toBeInTheDocument();
    });

    test('Если нажать на onClear, то вызовется метод onClear', () => {
        const onClear = jest.fn();
        const { getByTestId } = render(
            <Input onChange={console.log} onClear={onClear} value="value" />
        );

        expect(onClear).not.toHaveBeenCalled();

        fireEvent.click(getByTestId('onClearButton'));

        expect(onClear).toHaveBeenCalled();
    });

    test('Если передать isLoading, то появится loader', () => {
        const onClear = jest.fn();
        const { getByTestId } = render(
            <Input onChange={console.log} onClear={onClear} isLoading value="value" />
        );

        expect(getByTestId('loader')).toBeInTheDocument();
    });

    test('Если передать Icon, то компонент должен отобразиться в DOM', () => {
        const Icon = <div data-test-id="inputIcon">icon</div>;
        const { getByTestId } = render(<Input onChange={console.log} Icon={Icon} value="value" />);

        expect(getByTestId('inputIcon')).toBeInTheDocument();
    });
});
