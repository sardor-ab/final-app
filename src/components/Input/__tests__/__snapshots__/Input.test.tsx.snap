// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Если есть helperText, то строка должна отобразиться в DOM 1`] = `
<div
  data-test-id="helperText"
  title="helperText"
>
  <span>
    helperText
  </span>
</div>
`;

exports[`Input Если есть label, то строка должна отобразиться в DOM 1`] = `
<input
  autocomplete="off"
  class=""
  data-test-id="input"
  id="input"
  value="value"
/>
`;

exports[`Input Снимок input-a 1`] = `
<input
  autocomplete="off"
  class=""
  data-test-id="input"
  id="input"
  value="value"
/>
`;
